src/EnergyPlus/IceThermalStorage.cc

```diff
@@ -1368,6 +1448,113 @@ namespace IceThermalStorage {
         this->ParasiticElecEnergy = 0.0;
     }
 
+    void DetailedIceStorageData::size(EnergyPlusData &state)
+    {
+        int const TESTankIndex = Util::FindItemInList(this->Name, state.dataIceThermalStorage->DetailedIceStorage, &DetailedIceStorageData::Name);
+        int const TESSizingIndex = state.dataIceThermalStorage->DetailedIceStorage(TESTankIndex).TESSizingIndex;
+        if (TESSizingIndex == 0) {
+            return;
+        }
+        std::string_view const tankType = "ThermalStorage:Ice:Detailed";
+        std::string_view const callingRoutine = "DetailedIceStorageData::size";
+        Real64 constexpr tankHeatOfFusion = 334000.0; // J/Kg
+        auto &plntLoop = state.dataPlnt->PlantLoop(this->plantLoc.loopNum);
+        int PltSizNum = plntLoop.PlantSizNum;
+        auto &plntSizData = state.dataSize->PlantSizData(PltSizNum);
+
+        int startPeak = state.dataIceThermalStorage->ThermalStorageSizing(TESSizingIndex).onPeakStart * state.dataGlobal->TimeStepsInHour;
+        int endPeak = state.dataIceThermalStorage->ThermalStorageSizing(TESSizingIndex).onPeakEnd * state.dataGlobal->TimeStepsInHour;
+        Real64 sizingFactor = state.dataIceThermalStorage->ThermalStorageSizing(TESSizingIndex).sizingFactor;
+        Real64 onPeakSumWaterFlow = 0.0;
+        Real64 offPeakSumWaterFlow = 0.0;
+        if (!plntLoop.plantDesWaterFlowRate.empty()) {
+            for (int ts = 0; ts < 24 * state.dataGlobal->TimeStepsInHour; ++ts) {
+                if (ts > startPeak && ts <= endPeak) {
+                    onPeakSumWaterFlow += plntLoop.plantDesWaterFlowRate[ts] / state.dataGlobal->TimeStepsInHour;
+                } else {
+                    offPeakSumWaterFlow += plntLoop.plantDesWaterFlowRate[ts] / state.dataGlobal->TimeStepsInHour;
+                }
+            }
+        }
+        Real64 Cp = plntLoop.glycol->getSpecificHeat(state, plntSizData.ExitTemp, callingRoutine);
+        Real64 rho = plntLoop.glycol->getDensity(state, plntSizData.ExitTemp, callingRoutine);
+        Real64 onPeakEnergy =
+            onPeakSumWaterFlow * rho * Cp * plntSizData.DeltaT * Constant::rSecsInHour; // need Joules here, J = m3/s * kg/m3 * J/kg-C * C * sec
```

Comment:
To calculate onPeakEnergy, onPeakSumWaterFlow is summed (1471 to 1477) from start to end of on peak period. This means that any demand side component that will be served by this tank needs to log it's time step load See epluspsz.csv. Then converted to joules. This is the amount of energy (accounting for the sizing factor) that the tank needs to meet. The time step water flow rate is estimated in CoolingWaterFlowSizing.cc, via calcCoilWaterFlowRates and stored in state.dataPlnt->PlantLoop(loopNum).compDesWaterFlowRate[coilNumOnPlant].tsDesWaterFlowRate[timeStep]. Those data are summed into plntLoop.plantDesWaterFlowRate[ts]. The sum is used to find the plant load via line 1481. From here the tank size is calculated at line 1491.  kWh for the detailed model and J for the simple model.